Ports/build_all.sh

```diff
@@ -1,89 +1,127 @@
 #!/usr/bin/env bash
 
-clean=false
-verbose=false
+LIGHT_GREEN='\033[1;32m'
+LIGHT_WHITE='\033[1m'
+LIGHT_RED='\033[1;31m'
+LIGHT_CYAN='\033[1;36m'
+LIGHT_YELLOW='\033[1;33m'
+RESET_COLOR='\033[0m'
 
-case "$1" in
-    clean)
-        clean=true
-        ;;
-    verbose)
-        verbose=true
-        ;;
-    *)
-        ;;
-esac
+some_failed=false
+action='build'
+verbose=false
+failfast=false
 
-case "$2" in
-    clean)
-        clean=true
-        ;;
-    verbose)
-        verbose=true
-        ;;
-    *)
-        ;;
-esac
+for arg in "$@"; do
+    case "$arg" in
+        clean*)
+            action="$arg"
+            ;;
+        verbose)
+            verbose=true
+            ;;
+        failfast)
+            failfast=true
+            ;;
+    esac
+done
 
 some_failed=false
-built_ports=""
+processed_ports=()
 
-for file in *; do
-    if [ -d $file ]; then
-        pushd $file > /dev/null
-            port=$(basename $file)
-            port_built=0
-            for built_port in $built_ports; do
-                if [ "$built_port" = "$port" ]; then
-                    port_built=1
-                    break
-                fi
-            done
-            if [ $port_built -eq 1 ]; then
-                echo "Built $port."
-                popd > /dev/null
-                continue
-            fi
-            if ! [ -f package.sh ]; then
-                echo "ERROR: Skipping $port because its package.sh script is missing."
-                popd > /dev/null
-                continue
-            fi
+: "${SERENITY_TOOLCHAIN:=Clang}"
+if [[ "$SERENITY_TOOLCHAIN" == "GNU" ]]; then
+    SERENITY_TOOLCHAIN=""
+fi
+
+: "${SERENITY_ARCH:="$(uname -m)"}"
```

Comment:
Are these the defaults we have other places? I would have thought `GNU` and `x86_64` would be preferred. Especially if you run from an Apple Silicon macOS machine.

Reply:
It is not required. I learnt shell-check doesn't report unused variable for all caps letter. These are not required actually, so I have removed them.